File: tests/fixtures/java_bug/Buggy.java
// Buggy Java — NullPointerException
public class Buggy {
    public static double calculateRatio(Integer numerator, Integer denominator) {
        if (denominator == 0) {
            return 0;
        }
        return numerator / denominator;  // BUG: numerator may be null → NullPointerException
    }

    public static void main(String[] args) {
        double result = calculateRatio(null, 2);  // BUG: passing null
        System.out.println("Result: " + result);
    }
}

File: tests/fixtures/rust_bug/buggy.rs
// Buggy Rust — unwrap on None
fn calculate_ratio(numerator: Option<i32>, denominator: i32) -> f64 {
    if denominator == 0 {
        return 0.0;
    }
    let num = numerator.unwrap();  // BUG: numerator is None → panic
    num as f64 / denominator as f64
}

fn main() {
    let result = calculate_ratio(None, 2);  // BUG: passing None
    println!("Result: {}", result);
}
